### DISCUSSION TITLE
An exclusive gleam shell

### DISCUSSION THREAD

**Discussion by suraj-mandal** [gleam-lang/gleam#1989] [Ideas & suggestions]
Hello,

I would very much like if the shell that is opened by the command `gleam shell` could open a shell where we can type gleam code instead of `erlang code`.

Speaking from user point of view, learning a whole new language just to operate the shell might be an inconvenience to some people.

I was having some trouble since I was using gleam and have not touched the erlang, so had to keep documentation of both the languages handy in order to operate both the code and the shell.

Also I felt somewhat difficult into incorporating the methods present in gleam standard library into the gleam shell. I mean I would have to find the erlang counterpart of the similar function that was present in the gleam standard library in order to get the result.

For example:

I found it difficult in executing the `capitalise` function in `gleam/string` module.

Is there any workaround since I am kinda new. However, if there isn't is there a way such that automatic translations of gleam code to erlang code might happen under the hood of the shell, so that I will be writing gleam code in the shell and get the result.

**Comment by lpil** [comment]
Hey @suraj-mandal ! I agree, it would be fab to have a Gleam shell. This would be a good project for someone

**Reply by suraj-mandal** [reply]
I will try out in the meantime.. I am learning glean as well. Thanks!
